lib/identity_config.rb

```diff
@@ -10,7 +10,20 @@ class << self
   end
 
   CONVERTERS = {
-    string: proc { |value| value.to_s },
+    # Allows loading a string configuration from a system environment variable
+    # ex: To read DATABASE_HOST from system environment for the database_host key
+    # database_host: ['env', 'DATABASE_HOST']
+    # To use a string value directly, you can specify a string explicitly:
+    # database_host: 'localhost'
+    string: proc do |value|
+      if value.is_a?(Array) && value.length == 2 && value.first == 'env'
+        ENV.fetch(value[1])
+      elsif value.is_a?(String)
+        value
+      else
+        raise 'invalid system environment configuration value'
+      end
+    end,
```

Comment:
I still feel like, I would want the part where we label this as coming from the ENV to be like, the *source* not the *type*? 

But given that we're parsing it like, this, small suggestion to use array destructuring to avoid hardcoding array index values?
```suggestion
    string: proc do |(value_or_env, env_key)|
      if env_key && value_or_env == 'env'
        ENV.fetch(env_key)
      elsif (value = value_or_env).present?
        value
      else
        raise 'invalid system environment configuration value'
      end
    end,
```

Reply:
>I still feel like, I would want the part where we label this as coming from the ENV to be like, the source not the type?

I'm not sure I understand, can you say more?

Reply:
I think I forgot where in the chain these procs get called, we can just forget about that comment lol